packages/office-ui-fabric-react/src/components/CommandBar/examples/CommandBar.Lazy.Example.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { CommandBar, ICommandBarItemProps } from 'office-ui-fabric-react/lib/CommandBar';
+import { useConstCallback } from '@uifabric/react-hooks';
+import { ContextualMenuItemType } from '../../ContextualMenu';
+
+export const CommandBarLazyExample: React.FunctionComponent = () => {
+  const [menuItems, setMenuItems] = React.useState<ICommandBarItemProps[] | undefined>(undefined);
+
+  const onMenuDismissed = useConstCallback(() => {
+    setMenuItems(undefined);
+  });
+
+  const timeoutRef = React.useRef<NodeJS.Timeout>();
+
+  const loadItems = useConstCallback(() => {
+    const itemCount = Math.floor(Math.random() * 5) + 1;
+
+    const newMenuItems: ICommandBarItemProps[] = [];
+
+    for (let i = 0; i < itemCount; i++) {
+      newMenuItems.push({
+        key: `sub-item-${i}`,
+        name: `Item ${i}`,
+      });
+    }
+
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+
+    timeoutRef.current = setTimeout(() => {
+      setMenuItems(newMenuItems);
+    }, 2000);
+  });
+
+  const items: ICommandBarItemProps[] = [
```

Comment:
Shouldn't these be memoized in some way to avoid re-creating the array every time? Or is that the point of the example? (If so you should call it out in a comment.)